src/float.cr

```diff
@@ -148,8 +148,17 @@ struct Float32
     Printer.print(self, io)
   end
 
+  private def hash_normalize
+    t = to_i32
+    t == self ? t : unsafe_as(Int32)
```

Comment:
It is neccessary cause of definition of `hash` function: "if a==b then hash(a)==hash(b)".

Without it `Hash(Int32|Fliat32, V)` is broken.